201-front-door-standard-premium/main.bicep

```diff
@@ -0,0 +1,137 @@
+@description('The name of the Front Door endpoint to create. This must be globally unique.')
+param endpointName string = 'afd-${uniqueString(resourceGroup().id)}'
+
+@description('The name of the SKU to use when creating the Front Door profile.')
+@allowed([
+  'Standard_AzureFrontDoor'
+  'Premium_AzureFrontDoor'
+])
+param skuName string = 'Standard_AzureFrontDoor'
+
+@description('The host name that should be used when connecting from Front Door to the origin.')
+param originHostName string
+
+var profileName = 'MyFrontDoor'
+var originGroupName = 'MyOriginGroup'
+var originName = 'MyOrigin'
+var routeName = 'MyRoute'
+
+resource profile 'Microsoft.Cdn/profiles@2020-09-01' = {
+  name: profileName
+  location: 'global'
+  sku: {
+    name: skuName
+  }
+}
+
+resource endpoint 'Microsoft.Cdn/profiles/afdEndpoints@2020-09-01' = {
+  name: endpointName
+  parent: profile
+  location: 'global'
+  properties: {
+    originResponseTimeoutSeconds: 240
+    enabledState: 'Enabled'
+  }
+}
+
+resource originGroup 'Microsoft.Cdn/profiles/originGroups@2020-09-01' = {
+  name: originGroupName
+  parent: profile
+  properties: {
+    loadBalancingSettings: {
+      sampleSize: 4
+      successfulSamplesRequired: 3
+    }
+    healthProbeSettings: {
+      probePath: '/'
+      probeRequestType: 'HEAD'
+      probeProtocol: 'Http'
+      probeIntervalInSeconds: 100
+    }
+  }
+}
+
+resource origin 'Microsoft.Cdn/profiles/originGroups/origins@2020-09-01' = {
+  name: originName
+  parent: originGroup
+  properties: {
+    hostName: originHostName
+    httpPort: 80
+    httpsPort: 443
+    originHostHeader: originHostName
+    priority: 1
+    weight: 1000
+  }
+}
+
+resource route 'Microsoft.Cdn/profiles/afdEndpoints/routes@2020-09-01' = {
+  name: routeName
+  parent: endpoint
+  dependsOn:[
+    origin // This explicit dependency is required to ensure that the origin group is not empty when the route is created.

```

Comment:
not sure I follow this one - how the route needs the the origin when there doesn't appear to be any relationship between them...

Reply:
Yes, this is a known issue in the Front Door RP. You can't deploy a route until you have an origin group with an origin in it. I'm in discussions with the PG about trying to fix this :)

Reply:
FYI this is the case on all of the samples. If the PG does fix it, I will update them all.